api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,309 @@
+// Exactly one of the fields must be defined.
+type AdminNetworkPolicyPorts struct {
+	// AllPorts cannot be "false" when it is set
+	// AllPorts allows the user to select all ports for all protocols, thus not
+	// selecting traffic based on L4 principles.
+	// If "true" then all ports are selected for all protocols.
+	// +optional
+	AllPorts *bool `json:"allPorts,omitempty"`
+
+	// The list of ports to allow/deny/pass traffic on, each item in this list is
+	// combined using a logical OR. When this field is present it should contain at
+	// least one item, and this rule allows/denies/passes traffic only if the traffic
+	// matches at least one port in the list.
+	// +optional
+	List []AdminNetworkPolicyPort `json:"list,omitempty"`
+}
+
+// AdminNetworkPolicyPort describes a port to select
+type AdminNetworkPolicyPort struct {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match. If not specified, this
+	// field defaults to TCP.
+	// +optional
+	// +kubebuilder:default=TCP
+	Protocol *v1.Protocol `json:"protocol,omitempty"`
+
+	// The port on the given protocol. This can either be a numerical or named
+	// port on a pod. If this field is not provided, this matches no port names and
+	// numbers.
+	// If present, only traffic on the specified protocol AND port will be matched.
+	// +optional
+	Port *intstr.IntOrString `json:"port,omitempty"`
+
+	// If set, indicates that the range of ports from port to endPort, inclusive,
+	// should be allowed by the policy. This field cannot be defined if the port field
+	// is not defined or if the port field is defined as a named (string) port.
+	// The endPort must be equal or greater than port.
+	// +optional
+	EndPort *int32 `json:"endPort,omitempty"`
+}
+
+// AdminNetPolRuleAction string describes the AdminNetworkPolicy action type.
+// +enum
+type AdminNetPolRuleAction string
+
+const (
+	// RuleActionPass enables admins to provide exceptions to ClusterNetworkPolicies and delegate this rule to
+	// K8s NetworkPolicies.
+	AdminNetPolRuleActionPass AdminNetPolRuleAction = "Pass"
+	// RuleActionDeny enables admins to deny specific traffic.
+	AdminNetPolRuleActionDeny AdminNetPolRuleAction = "Deny"
+	// RuleActionAllow enables admins to specifically allow certain traffic.
+	AdminNetPolRuleActionAllow AdminNetPolRuleAction = "Allow"
+)
+
+// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from.
+// Exactly one of the selector pointers should be set for a given peer. If a
+// consumer observes none of its fields are set, they should assume an unknown
+// option has been specified and fail closed.
+type AdminNetworkPolicyPeer struct {
+	// Namespaces defines a way to select a set of Namespaces
+	// +optional
+	Namespaces *NamespaceSet `json:"namespaces,omitempty"`
+	// NamespacedPods defines a way to select a specific set of pods in
+	// in a set of namespaces
+	// +optional
+	NamespacedPods *NamespaceAndPodSet `json:"namespacedPods,omitempty"`
+}
```

Comment:
It is potentially confusing that this is almost but not quite the same as `AdminNetworkPolicySubject`.
Also, like there, it seems like maybe this could be replaced with a mandatory namespace specification and an optional pod specification.

Reply:
I'm going to point to @rahulkjoshi's previous comment here since I think it's even more applicable when describing an ANP peer. 

https://github.com/kubernetes-sigs/network-policy-api/pull/30/files#r846298297

Reply:
that link doesn't seem to point to any specific comment for me